### ISSUE TITLE
Unable to log into console after Tenant creation using Custom Domains

### ISSUE THREAD

**Issue by zulrang** [minio/operator#1245]
After provisioning a new Tenant, none of the credentials work to log into the Console for the Tenant.

## Expected Behavior
Credentials should auth correctly

## Current Behavior
When provisioning a Tenant, I can provide "additional users" and the credentials correctly show up in the `<tenant>-user-0` secret in the namespace.  But they don't work to log into the console.

The `<tenant>-secret` secret is empty:
```
apiVersion: v1
data:
  accesskey: ""
  secretkey: ""
```

The `<tenant>-env-configuration` has auto-generated credentials that don't work either.  These are reflected in the /tmp/minio/config.env within the pods.

Clicking the "console" button on the Operator console leads to a 403 error.

## Possible Solution
?

## Steps to Reproduce (for bugs)
1.  Using the minio plugin v4.4.28, run `kubectl minio init`
2.  Open and log into the Operator Console
3.  Create a tenant without exposing the services *using Custom Domains* and disable TLS
4.  Unable to log into the Tenant Console

## Context
Unable to use Minio at all

## Regression
Unknown.  Using v4.4.28

## Your Environment
Minio Operator v4.4.28
Kubernetes 1.21.2
Istio 1.14.1

Tenant manifest:

```
apiVersion: minio.min.io/v2
kind: Tenant
metadata:
  name: objects
  namespace: objects
spec:
  configuration:
    name: objects-env-configuration
  credsSecret:
    name: objects-secret
  exposeServices: {}
  image: minio/minio:RELEASE.2022-08-13T21-54-44Z
  imagePullSecret: {}
  log:
    audit:
      diskCapacityGB: 5
    db:
      resources: {}
      securityContext:
        fsGroup: 999
        runAsGroup: 999
        runAsNonRoot: true
        runAsUser: 999
      volumeClaimTemplate:
        metadata:
          name: objects-log
        spec:
          accessModes:
          - ReadWriteOnce
          resources:
            requests:
              storage: "5368709120"
        status: {}
    resources: {}
    securityContext:
      fsGroup: 1000
      runAsGroup: 1000
      runAsNonRoot: true
      runAsUser: 1000
  mountPath: /export
  pools:
  - affinity:
      podAntiAffinity:
        requiredDuringSchedulingIgnoredDuringExecution:
        - labelSelector:
            matchExpressions:
            - key: v1.min.io/tenant
              operator: In
              values:
              - objects
            - key: v1.min.io/pool
              operator: In
              values:
              - pool-0
          topologyKey: kubernetes.io/hostname
    name: pool-0
    resources:
      requests:
        cpu: "1"
        memory: 8Gi
    servers: 1
    volumeClaimTemplate:
      metadata:
        name: data
      spec:
        accessModes:
        - ReadWriteOnce
        resources:
          requests:
            storage: "107374182400"
        storageClassName: direct-csi-min-io
      status: {}
    volumesPerServer: 1
  prometheus:
    diskCapacityGB: 5
    resources: {}
    securityContext:
      fsGroup: 1000
      runAsGroup: 1000
      runAsNonRoot: true
      runAsUser: 1000
  requestAutoCert: false
  users:
  - name: objects-user-0
status:
  availableReplicas: 1
  certificates:
    autoCertEnabled: false
  currentState: Initialized
  drivesOnline: 1
  healthStatus: green
  pools:
  - legacySecurityContext: false
    ssName: objects-pool-0
    state: PoolInitialized
  provisionedUsers: true
  revision: 0
  syncVersion: v4.3.0
  usage:
    capacity: 105135398912
    rawCapacity: 107374182400
    rawUsage: 64184320
  writeQuorum: 1
```

**Comment by zulrang** [comment]
After deleting all objects and namespaces related to the tenants and minio-operator, the rerunning `kubectl minio init` this issue was solved.

**Comment by zulrang** [comment]
Correction: it works as long as I don't set Custom Domains.

The following Tenant will NOT allow you to log into the console at all - it seems there's some kind of communication breakdown between the Console and Minio:

```
apiVersion: v1
items:
- apiVersion: minio.min.io/v2
  kind: Tenant
  metadata:
    name: buckets
    namespace: buckets
  scheduler:
    name: ""
  spec:
    configuration:
      name: buckets-env-configuration
    credsSecret:
      name: buckets-secret
    exposeServices: {}
    features:
      domains:
        console: https://buckets-console.mydomain.com/
        minio:
        - https://buckets.mydomain.com/
    image: minio/minio:RELEASE.2022-08-22T23-53-06Z
    imagePullSecret: {}
    log:
      audit:
        diskCapacityGB: 5
      db:
        resources: {}
        securityContext:
          fsGroup: 999
          runAsGroup: 999
          runAsNonRoot: true
          runAsUser: 999
        volumeClaimTemplate:
          metadata:
            name: buckets-log
          spec:
            accessModes:
            - ReadWriteOnce
            resources:
              requests:
                storage: "5368709120"
          status: {}
      resources: {}
      securityContext:
        fsGroup: 1000
        runAsGroup: 1000
        runAsNonRoot: true
        runAsUser: 1000
    mountPath: /export
    pools:
    - affinity:
        podAntiAffinity:
          requiredDuringSchedulingIgnoredDuringExecution:
          - labelSelector:
              matchExpressions:
              - key: v1.min.io/tenant
                operator: In
                values:
                - buckets
              - key: v1.min.io/pool
                operator: In
                values:
                - pool-0
            topologyKey: kubernetes.io/hostname
      name: pool-0
      resources:
        requests:
          cpu: "1"
          memory: 4Gi
      servers: 4
      volumeClaimTemplate:
        metadata:
          name: data
        spec:
          accessModes:
          - ReadWriteOnce
          resources:
            requests:
              storage: "68719476736"
          storageClassName: direct-csi-min-io
        status: {}
      volumesPerServer: 4
    prometheus:
      diskCapacityGB: 5
      resources: {}
      securityContext:
        fsGroup: 1000
        runAsGroup: 1000
        runAsNonRoot: true
        runAsUser: 1000
    requestAutoCert: false
    users:
    - name: buckets-user-0
  status:
    availableReplicas: 4
    certificates:
      autoCertEnabled: false
    currentState: Initialized
    drivesOnline: 16
    healthStatus: green
    pools:
    - legacySecurityContext: false
      ssName: buckets-pool-0
      state: PoolInitialized
    provisionedUsers: true
    revision: 0
    syncVersion: v4.3.0
    usage:
      capacity: 536699305984
      rawCapacity: 1099511627776
      rawUsage: 910819328
      usage: 5
    writeQuorum: 9
kind: List
metadata:
  resourceVersion: ""
  selfLink: ""
```

So if I try to log into the console through the Operator console by clicking on "Console", I get a blank page and the operator console logs says the following:
`2022/08/24 18:36:23 couldn't login to tenant and get cookie`

If I try to access the minio console through the domain name, the page comes up but won't log in:
![image](https://user-images.githubusercontent.com/1043189/186500985-1ae10031-29d2-47ed-b28e-4cc97e838274.png)

No errors in the tenant console log:
![image](https://user-images.githubusercontent.com/1043189/186501096-0d4b9141-63d7-4a1f-a733-93357645655a.png)

**However, `mc` works just fine, as it should, using the same exact domain and credentials.**

The request log for the Console login:
```
{"response_code":401,"request_id":"85df9ed9-75e9-4716-9119-fcd4e1ce08da","connection_termination_details":null,"x_forwarded_for":"x.x.x.x","user_agent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/x.x.x.x Safari/537.36 Edg/x.x.x.x","upstream_transport_failure_reason":null,"response_code_details":"via_upstream","duration":21,"bytes_received":58,"downstream_local_address":"x.x.x.x:8443","start_time":"2022-08-24T18:44:08.239Z","requested_server_name":"buckets-console.mydomain.com","route_name":null,"path":"/api/v1/login","downstream_remote_address":"x.x.x.x:65497","response_flags":"-","upstream_host":"x.x.x.x:9090","upstream_service_time":"21","method":"POST","authority":"buckets-console.mydomain.com","bytes_sent":73,"upstream_local_address":"x.x.x.x:45800","upstream_cluster":"outbound|9090||buckets-console.buckets.svc.cluster.local","protocol":"HTTP/2"}
```

And a `mc ls` request:
```
{"downstream_local_address":"x.x.x.x:8443","route_name":null,"authority":"buckets.mydomain.com","downstream_remote_address":"x.x.x.x:53644","user_agent":"MinIO (linux; amd64) minio-go/v7.0.34 mc/RELEASE.2022-08-11T00-30-48Z","protocol":"HTTP/1.1","upstream_service_time":"7","request_id":"8400ef6d-ad83-4d40-aec1-930c2ba660c6","response_code":200,"duration":7,"requested_server_name":"buckets.mydomain.com","start_time":"2022-08-24T18:47:09.531Z","path":"/","bytes_received":0,"bytes_sent":275,"upstream_local_address":"x.x.x.x:38852","method":"GET","upstream_host":"x.x.x.x:9000","upstream_transport_failure_reason":null,"response_flags":"-","upstream_cluster":"outbound|80||minio.buckets.svc.cluster.local","connection_termination_details":null,"x_forwarded_for":"x.x.x.x","response_code_details":"via_upstream"}
```

Note: if I set up a tenant WITHOUT custom domains from this same Operator, it works fine.

**Comment by Rezkmike** [comment]
Im having the same problem as @zulrang. Once set the console domain for the bucket, i will not able to login and open manage dashboard from the operator.

**Comment by cowjen01** [comment]
@Rezkmike I'm new to the MinIO Operator, but from my previous knowledge of MinIO this issue can be solved by setting the correct value of `MINIO_SERVER_URL` variable.

**Comment by Rezkmike** [comment]
Now its working on me, after I disabled the TLS and using custom domain. If I turn on the TLS, the configured custom domain will not be accessible.

**Comment by rufreakde** [comment]
> LS and using custom domain

@Rezkmike @zulrang  I see in your example you use istio. How did you disable the TLS?  maybe with:
```yaml
        operator:
          replicaCount: 2
          env:
            - name: MINIO_OPERATOR_TLS_ENABLE
              value: "off"
            - name: MINIO_CONSOLE_TLS_ENABLE
              value: "off"
```
on operator side

```yaml
  requestAutoCert: true
  features:
    bucketDNS: false
```
on tenant side? 

Was this enough? We want to use istio to handle the TLS.

**Comment by Rezkmike** [comment]
I forgot how to turn off the TLS, probably need to turn off the "requestAutoCert" if im not mistaken. 

But here is the latest configuration that working on my setup. TLS handled by Nginx Ingress 

```bash
certificate:
    ## Use this field to provide one or more external CA certificates. This is used by MinIO
    ## to verify TLS connections with other applications:
    ## https://github.com/minio/minio/tree/master/docs/tls/kubernetes#2-create-kubernetes-secret
    externalCaCertSecret: [ ]
    ## Use this field to provide a list of Secrets with external certificates. This can be used to configure
    ## TLS for MinIO Tenant pods. Create secrets as explained here:
    ## https://github.com/minio/minio/tree/master/docs/tls/kubernetes#2-create-kubernetes-secret
    externalCertSecret: [ ]
    ## Enable automatic Kubernetes based certificate generation and signing as explained in
    ## https://kubernetes.io/docs/tasks/tls/managing-tls-in-a-cluster
    requestAutoCert: false
    ## This field is used only when "requestAutoCert" is set to true. Use this field to set CommonName
    ## for the auto-generated certificate. Internal DNS name for the pod will be used if CommonName is
    ## not provided. DNS name format is *.minio.default.svc.cluster.local
    certConfig: { }
  ## MinIO features to enable or disable in the MinIO Tenant
  ## https://github.com/minio/operator/blob/master/docs/crd.adoc#features
  features:
    bucketDNS: true
    # domains: { }
    domains:
      console: https://minio2-console.<FQDN>
      minio:
      - https://minio2-api.<FQDN>
```

**Comment by pjuarezd** [comment]
I think most of the communication issues related to TLS are already resolved, please try operator v5.0.0 and above and let us know if still having problem.

**Comment by pjuarezd** [comment]
Closing this for now, let us know if stil having issues in latest version
